tools/scylla-sstable-scripts/dump.lua

```diff
@@ -0,0 +1,257 @@
+        write_ttl(cell)
+    end
+    if not cell.is_live then
+        writer:key("deletion_time");
+        writer:string(tostring(cell.deletion_time));
+    end
+end
+
+function write_collection(cell)
+    if cell.tombstone then
+        writer:key("tombstone")
+        write_tombstone(cell.tombstone)
+    end
+    writer:key("cells")
+    writer:start_array()
+    for _, v in ipairs(cell.values) do
+        writer:start_object();
+
+        writer:key("key")
+        writer:string(v.key)
+
+        writer:key("value")
+        writer:start_object()
+        write_atomic_cell(v.value)
+        writer:end_object()
+
+        writer:end_object()
+    end
+    writer:end_array()
+end
+
+function write_cells(cells)
+    writer:start_object();
+
+    for name, cell in pairs(cells) do
+        writer:key(name);
+        writer:start_object();
+
+        if cell.is_atomic == true then
+            write_atomic_cell(cell)
+        else
+            write_collection(cell)
+        end
+
+        writer:end_object();
+    end
+
+    writer:end_object();
+end
+
+function on_start_of_stream()
+    writer:start_object();
+    writer:key("sstables");
+    writer:start_object();
+end
+
+function on_new_sstable(sst)
+    if sst == nil then
+        writer:key("anonymous");
+    else
+        writer:key(sst.filename)
+    end
+    writer:start_array();
+end
+
+function consume_partition_start(ps)
+    writer:start_object();
+
+    clustering_array_created = false
+
+    writer:key("key");
+    write_key(ps)
+
+    if ps.tombstone then
+        writer:key("tombstone");
+        write_tombstone(ps.tombstone)
+    end
+end
+
+function consume_static_row(sr)
+    writer:key("static_row");
+    write_cells(sr.cells);
+end
+
+function consume_clustering_row(cr)
+    maybe_start_clustering_array()
+
+    writer:start_object();
+
+    writer:key("type");
+    writer:string("clustering-row")
+
+    writer:key("key");
+    write_key(cr)
+
+    if cr.tombstone then
+        writer:key("tombstone");
+        write_tombstone(cr.tombstone);
+        writer:key("shadowable_tombstone");
+        write_tombstone(cr.shadowable_tombstone);
+    end
+
+    if cr.marker then
+        writer:key("marker");
+        writer:start_object();
+
+        writer:key("timestamp");
+        writer:int(cr.marker.timestamp);
+
+        if cr.marker.is_live and cr.marker.has_ttl then
+            write_ttl(cr.marker)
+        end
+
+        writer:end_object();
+    end
+
+    writer:key("columns");
+    write_cells(cr.cells);
+
+    writer:end_object();
+end
+
+function consume_range_tombstone_change(crt)
+    maybe_start_clustering_array()
+
+    writer:start_object();
+
+    writer:key("type");
+    writer:string("range-tombstone-change");
+
+    if crt.key then
+        writer:key("key");
+        write_key(crt);
+    end
+
+    writer:key("weight");
+    writer:int(crt.key_weight);
+
+    writer:key("tombstone");
+    write_tombstone(crt.tombstone);
+
+    writer:end_object();
+end
+
```

Comment:
I somehow expected that this will be awk-like, with an empty script printing everything. Perhaps we should expose a "print_range_tombstone_change" function (and equivalents for other types) or even just print(something) that overloads based on the type, so that this script can be just calls to print(). Not a blocker in any case.

Reply:
I can add support for `tostring()` operator for each of the fragment types, although I think this should use the default `operator<<` for these elements.
I guess we can also expose a method to print the fragments as json, we have the json conversion code ready in C++ already after all.